src/compiler/checker.ts

```diff
@@ -6361,11 +6361,10 @@ namespace ts {
          * @param typeParameters The requested type parameters.
          * @param minTypeArgumentCount The minimum number of required type arguments.
          */
-        function fillMissingTypeArguments(typeArguments: Type[] | undefined, typeParameters: TypeParameter[] | undefined, minTypeArgumentCount: number, location?: Node) {
+        function fillMissingTypeArguments(typeArguments: Type[] | undefined, typeParameters: TypeParameter[] | undefined, minTypeArgumentCount: number, isJavaScript?: boolean) {

```

Comment:
Probably better to just make `isJavascript` required (same for other one)

Reply:
I agree that required parameters are generally better; I'm not convinced that they're an improvement for `fillMissingTypeArguments`. There are 7 calls, 2 of which are so deep in instantiation that they have to call `isInJavascriptFile` on a node that may or may not be relevant. One of the 6 calls to `getSignatureInstantiation` is like this too.

I pushed the change so we can look at it and decide if it's worthwhile.